src/test/java/edu/hm/hafner/analysis/ast/AbstractAstTest.java

```diff
@@ -0,0 +1,208 @@
+    static final String LINE93_INCR = "EXPR POST_INC IDENT SEMI ";
+    static final String LINE94_CALL = "EXPR METHOD_CALL DOT IDENT IDENT ELIST EXPR METHOD_CALL IDENT ELIST RPAREN RPAREN SEMI ";
+    static final String LINE95_CALL = "EXPR METHOD_CALL IDENT ELIST RPAREN SEMI ";
+    static final String LINE96_RCURLY = "RCURLY ";
+    static final String LINE97_RCURLY = "RCURLY ";
+    static final String LINE98_ELSE = "LITERAL_ELSE SLIST ";
+    static final String LINE99_BREAK = "LITERAL_BREAK SEMI ";
+    static final String LINE100_RCURLY = "RCURLY ";
+    static final String LINE101_RCURLY = "RCURLY ";
+    static final String LINE103_RETURN = "LITERAL_RETURN EXPR IDENT SEMI ";
+    static final String LINE104_RCURLY = "RCURLY ";
+    static final String LINE105_RCURLY = "RCURLY ";
+
+    static final String WHOLE_METHOD = LINE67_METHOD + LINE68_VAR + LINE69_VAR
+            + LINE70_VAR + LINE71_VAR + LINE73_VAR + LINE75_VAR + LINE76_IF + LINE77_ASSIGN + LINE78_RCURLY
+            + LINE79_ELSE
+            + LINE80_ASSIGN + LINE81_RCURLY + LINE82_FOR + LINE83_IF + LINE84_IF + LINE85_ASSIGN + LINE86_RCURLY
+            + LINE87_ELSE + LINE88_ASSIGN + LINE89_RCURLY
+            + LINE90_CALL + LINE91_CALL + LINE92_IF + LINE93_INCR + LINE94_CALL + LINE95_CALL + LINE96_RCURLY
+            + LINE97_RCURLY + LINE98_ELSE + LINE99_BREAK + LINE100_RCURLY
+            + LINE101_RCURLY + LINE103_RETURN + LINE104_RCURLY;
+    static final String WHOLE_CLASS = LINE14_CLASS + LINE16_FIELD + LINE18_FIELD
+            + LINE25_CTOR + LINE26_SUPER + LINE27_RCURLY + LINE29_METHOD
+            + LINE30_RETURN + LINE31_RCURLY + LINE33_METHOD + LINE34_RETURN + LINE35_RCURLY
+            + WHOLE_METHOD + LINE105_RCURLY;
+
+    Ast createAst(final int lineNumber) {
+        String fileName = createJavaSourceTemporaryFile("elements.ast-test");
+
+        return createAst(fileName, lineNumber);
+    }
+
+    /**
+     * Creates the AST under test for the specified file at the given line number.
+     *
+     * @param fileName
+     *         the file to scan
+     * @param lineNumber
+     *         the first interesting line number
+     *
+     * @return the created {@link Ast}
+     */
+    protected abstract Ast createAst(String fileName, int lineNumber);
+
+    void assertThatAstIs(final Ast ast, final String expectedResult) {
+        String realResult = ast.chosenAreaAsString(' ');
+
+        assertThat(expectedResult).isNotNull();
+        assertThat(expectedResult).isEqualTo(realResult);
+    }
+
+    String createJavaSourceTemporaryFile(final String fileName) {
+        File warnings = createCopyInTemp(fileName);
+        return warnings.getAbsolutePath();
+    }
+
+    private File createCopyInTemp(final String fileName) {
+        try {
+            File sourceCode = File.createTempFile("ast", ".java");
+            sourceCode.deleteOnExit();
+
+            InputStream stream = AbstractAstTest.class.getResourceAsStream(fileName);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Ensure that resources like this InputStream object are closed after use](https://app.codacy.com/manual/uhafner/analysis-model/pullRequest?prid=5019963)